### PR TITLE
[sc 254 - fixed] re-add match syntax on adts to hvm lang

### PR DESCRIPTION
**Pull request by LunaAmora**

### CODE REVIEW

File: src/term/term_to_net.rs [hunk 1]
```diff
@@ -143,20 +143,23 @@ fn encode_term(
...
+    Term::Match { scrutinee, arms } => {
       let if_ = inet.new_node(Mat);
 
-      let cond = encode_term(inet, cond, Port(if_, 0), scope, vars, global_vars, label_generator);
+      let cond = encode_term(inet, scrutinee, Port(if_, 0), scope, vars, global_vars, label_generator);
       link_local(inet, Port(if_, 0), cond);
 
-      let sel = inet.new_node(Con);
-      inet.link(Port(sel, 0), Port(if_, 1));
+      if let Some((zero, succ)) = native_match(arms.clone()) {
```
Comment (developedby) [inline src/term/term_to_net.rs:152]: Instead of indenting, I think it would be better to have an `unwrap` here.

**Review by developedby — CHANGES_REQUESTED** [review]

File: src/term/mod.rs [hunk 2]
```diff
@@ -353,6 +361,106 @@ impl Term {
...
+          }
+
+          free_vars.extend(new_scope);
+        }
+      }
+      Term::Ref { .. } | Term::Num { .. } | Term::Era => {}
+    }
+  }
+
+  pub fn num_match(scrutinee: Term, zero_term: Term, succ_label: Option<Name>, mut succ_term: Term) -> Term {
```
[the remark this answers is not in the document]
Comment (developedby) [reply src/term/mod.rs:425]: You resolved, but didn't change anything here

**Review by developedby — COMMENTED** [review]
.

File: src/term/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (LunaAmora) [reply src/term/mod.rs:425]: Oh, sorry, the name is so ambiguous that i though it was one of the others i added the description. Will fix

**Pull request merged**